File: api/api-java/src/main/java/com/api_java/exception/CustomException.java
package com.api_java.exception;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CustomException extends RuntimeException {
  
  @JsonProperty("status_code")
  private final int statusCode;
  
  @JsonProperty("content")
  private final String content;

  public CustomException(String content, int statusCode) {
      super(content);
      this.content = content;
      this.statusCode = statusCode;
  }

  public int getStatusCode() {
      return statusCode;
  }

  public String getContent() {
      return content;
  }
}

File: api/api-nestjs/src/exception/custom.exception.ts
import { HttpException, HttpStatus } from '@nestjs/common';
import { ExceptionResponse } from './exception-response';

export class CustomException extends HttpException {
  constructor(statusCode: HttpStatus, content: string) {
    const response = new ExceptionResponse(statusCode, content);
    super(response, statusCode);
  }
}